server/auth_server.go

```diff
@@ -2,48 +2,35 @@ package server
 
 import (
 	"context"
-	"github.com/golang/protobuf/ptypes/empty"
+	"fmt"
 	"github.com/onepanelio/core/api"
 	v1 "github.com/onepanelio/core/pkg"
 	"github.com/onepanelio/core/server/auth"
 	"github.com/pkg/errors"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/metadata"
 	"google.golang.org/grpc/status"
-	"strings"
 )
 
-type AuthServer struct{}
-
-func NewAuthServer() *AuthServer {
-	return &AuthServer{}
+// AuthServer contains logic for checking Authorization of resources in the system
+type AuthServer struct {
+	fqdns map[string]string // map from namespace -> namespace.domain
 }
-func (a *AuthServer) IsWorkspaceAuthenticated(ctx context.Context, request *api.IsWorkspaceAuthenticatedRequest) (*empty.Empty, error) {
-	if ctx == nil {
-		return &empty.Empty{}, nil
-	}
-	client := ctx.Value("kubeClient").(*v1.Client)
-	md, ok := metadata.FromIncomingContext(ctx)
-	if !ok {
-		return &empty.Empty{}, errors.New("Error parsing headers.")
-	}
-	//Expected format: x-original-authority:[name--default.alexcluster.onepanel.io]
-	xOriginalAuth := md.Get("x-original-authority")[0]
-	fqdn := md.Get("fqdn")[0]
-	if xOriginalAuth == fqdn {
-		return &empty.Empty{}, nil
-	}
-	pos := strings.Index(xOriginalAuth, ".")
-	if pos == -1 {
-		return &empty.Empty{}, errors.New("Error parsing x-original-authority. No '.' character.")
+
+// NewAuthServer creates a new AuthServer with some pre-cached data.
+// - namespaces are all of the user namespaces in the system
+// - config is the system config.
+//     ONEPANEL_DOMAIN is required at a minimum.
+func NewAuthServer(namespaces []*v1.Namespace, config map[string]string) *AuthServer {
+	server := &AuthServer{
+		fqdns: make(map[string]string),
```

Comment:
Bit confused about this, why do we have multiple `fqdns` and why are namespaces involved. `fqdn` is the application domain in [params.yaml](https://github.com/onepanelio/manifests/blob/master/common/onepanel/base/vars.yaml#L50)

Reply:
Looks like you're setting the workspace names in here. Wouldn't it be better to just pass through auth for `fdqn` from `params.yaml`?

Reply:
@rushtehrani This is bad naming on my part. 

First - do you mean pass in the fqdn instead of the entire config? I'm up for that.

Second - this is supposed to be the suffix part of  `workspace.name--(suffix)`.

When we check permissions for workspace related items, it's a quick-ish way to get the expected full path of it -` workspace.name--namespace.domain`.

We prepend the `workspace.name`.